amy/dashboard/views.py

```diff
@@ -245,57 +245,135 @@ def training_progress(request):
 
 
 class GetInvolvedCreateView(LoginRequiredMixin, RedirectSupportMixin, AMYCreateView):
-    # permission_required = TODO
     model = TrainingProgress
     form_class = GetInvolvedForm
     template_name = "get_involved_form.html"
     success_url = reverse_lazy("training-progress")
     success_message = (
-        "Thank you. Your Get Involved submission will be reviewed within 7-10 days."
+        "Thank you. Your Get Involved submission will be evaluated within 7-10 days."
     )
 
+    def user_may_create_submission(self, user):
+        """The user may create a submission if either of the conditions below are met:
+        1. no progress exists for the Get Involved step for this user
+        2. all existing progress for the Get Involved step for this user has state "a"
+        """
+        get_involved = TrainingRequirement.objects.get(name="Get Involved")
+        existing_progresses = TrainingProgress.objects.filter(
+            requirement=get_involved, trainee=user
+        )
+        num_existing = existing_progresses.count()
+        if num_existing == 0:
+            return True
+        else:
+            num_asked_to_repeat = existing_progresses.filter(state="a").count()
+            if num_asked_to_repeat == num_existing:
+                return True
+            else:
+                return False
+
```

Comment:
This should become part of the form validation, for example in the `clean()` method. Combined with no use of `post()` to handle form valid/non-valid check (see my review of #2492), this will simplify the view.